src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml

```diff
@@ -101,6 +102,689 @@ tags:
+
+`affected_global_entities_filter` - allows to specify array of addresses. If specified response will contain transactions that affected all of given global entities. Affected global entities are defined by substates published as result of transaction. Each substate references entity, if that entity is global it's considered affected global entity, if it's not global then it's global ancestor is treated as affected global entity.
+
+`events_filter` - allows to filter transaction stream based on published events. Currently only deposit and withdrawal events are supported. 
+
+It's complex object where you can specify:
+- `event` - type of event (required)  i.e `{"event": "withdrawal"}` 
+- `emitter_address` - will return all transactions where given entity emitted event. i.e `{"emitter_address" : "consensusmanager_rdx1scxxxxxxxxxxcnsmgrxxxxxxxxx000999665565xxxxxxxxxcnsmgr"}` 
+- `resource_address` - only transactions deposit/withdrawal events for given resource will be returned i.e `{"resource_address" : "resource_rdx1tknxxxxxxxxxradxrdxxxxxxxxx009923554798xxxxxxxxxradxrd"}` 
+
+You can combine multiple event filters, it'll look for transactions that satisfies all conditions. i.e you can query for all transactions, where given entity emitted deposit event for specified resource:
+
+```
+{
+  "events_filter": [
+    {
+    "event": "Deposit",
+    "emitter_address": "consensusmanager_rdx1scxxxxxxxxxxcnsmgrxxxxxxxxx000999665565xxxxxxxxxcnsmgr",
+    "resource_address": "resource_rdx1tknxxxxxxxxxradxrdxxxxxxxxx009923554798xxxxxxxxxradxrd"
+    }
+  ]
+}
+```
+
+
+
+### Pagination
+
+If `limit_per_page` request parameter is lower than number of transactions to be returned, `next_cursor` will be returned. Use it in next request as `cursor` parameter to fetch further pages.
+
+i.e:
+```
+{
+  "kind_filter":"All",
+  "from_ledger_state": {"state_version": 100 },
+  "at_ledger_state": {"state_version": 300 },
+  "limit_per_page":1
+}
+```
+will return 
+```
+{
+...
+"next_cursor": "eyJ2IjoyOTl9",
+...
+}
+```
+
+use that in next query to fetch next pages:
+```
+{
+  "kind_filter":"All",
+  "from_ledger_state": {"state_version": 100 },
+  "at_ledger_state": {"state_version": 300 },
+  "limit_per_page":1,
+  "cursor": "eyJ2IjoyOTl9"
+}
+```
+
+### Ordering
+By default ordering in that endpoint is **descending** (highest state version first), so when using cursor you'll fetch newest first and with each page you'll fetch older transactions.
+
+You can change default behaviour by seting `order` parameter to `Asc`
+
+```
+{ 
+  ...
+  "order": "Asc" 
+  ...
+}
+```
+Keep in mind that if order is ascending you'll start fetching transactions from oldest first and with each page you'll fetch newer transactions.
+
+### Reading events to find out what happened in certain dApp.
+### TODO PP:
```

Comment:
we support only withdraw/deposit event. I could show how to filter using `affected_global_entities` filter here but it's based on substates so not sure if that's what we need here.